packages/peregrine/lib/List/items.js

```diff
@@ -1,104 +1,52 @@
-    const { selectionModel } = props;
-    let selection;
-
-    if (selectionModel === 'radio') {
-        selection = new Set().add(key);
-    }
-
-    if (selectionModel === 'checkbox') {
-        selection = new Set(prevState.selection);
-
-        if (selection.has(key)) {
-            selection.delete(key);
-        } else {
-            selection.add(key);
-        }
-    }
-
-    return { selection };
-});
-
-class Items extends Component {
-    static propTypes = {
-        getItemKey: PropTypes.func.isRequired,
-        items: iterable.isRequired,
-        renderItem: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
-        selectionModel: PropTypes.oneOf(['checkbox', 'radio'])
-    };
-
-    static defaultProps = {
-        getItemKey: ({ id }) => id,
-        selectionModel: 'radio'
-    };
-
-    state = {
-        cursor: null,
-        hasFocus: false,
-        selection: new Set()
-    };
-
-    render() {
-        const { getItemKey, items, renderItem } = this.props;
-        const { cursor, hasFocus, selection } = this.state;
-
-        const children = Array.from(items, (item, index) => {
-            const key = getItemKey(item, index);
-
-            return (
-                <ListItem
-                    key={key}
-                    item={item}
-                    itemIndex={index}
-                    render={renderItem}
-                    hasFocus={hasFocus && cursor === index}
-                    isSelected={selection.has(key)}
-                    onBlur={this.handleBlur}
-                    onClick={this.getClickHandler(key)}
-                    onFocus={this.getFocusHandler(index)}
-                />
-            );
-        });
-
-        return <Fragment>{children}</Fragment>;
-    }
-
-    syncSelection() {
-        const { selection } = this.state;
-        const { onSelectionChange } = this.props;
-
-        if (onSelectionChange) {
-            onSelectionChange(selection);
-        }
-    }
-
-    handleBlur = () => {
-        this.setState(removeFocus);
-    };
-
-    getClickHandler = memoize(key => () => {
-        this.setState(updateSelection(key), this.syncSelection);
-    });
-
-    getFocusHandler = memoize(index => () => {
-        this.setState(updateCursor(index));
+import Item from './item';
+
+const Items = props => {
+    const {
+        getItemKey,
+        items,
+        renderItem,
+        selectionModel,
+        onSelectionChange
+    } = props;
+    const [
+        { cursor, hasFocus, selection },
+        { setFocus, removeFocus, updateSelection }
+    ] = useListState({ selectionModel, onSelectionChange });
+    const itemsElements = items.map((item, index) => {
+        const key = getItemKey(item, index);
+        return (
+            <Item
+                key={key}
+                uniqueID={key}
+                item={item}
+                itemIndex={index}
+                render={renderItem}
+                hasFocus={hasFocus && cursor === key}
```

Comment:
[The previous version](https://github.com/magento/pwa-studio/pull/1515/files?file-filters%5B%5D=.js#diff-e45fdb12add04198a28a49de2c2d7b40L70) had this line as 

```jsx
hasFocus={hasFocus && cursor === index}
```

Did you change `index` to `key` intentionally here?

Reply:
Yeah, I made the change. It is supposed to be `key` since `cursor` is always set to the `key` of the `item` that is selected. It works in this case since the `key` is `index` but for other cases where `getItemKey` function returns something other than `index`, the component will fail.